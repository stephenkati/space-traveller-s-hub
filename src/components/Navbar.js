import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../Images/planet.png';

const Navbar = () => (
  <nav>
    <div className="app-logo">
      <img src={logo} alt="logo" className="logo" />
      <h1 className="app-name">Space Travelers&rsquo; Hub</h1>
    </div>
    <ul className="navLinks">
      <li>
        <NavLink to="/">
          Rockets
        </NavLink>
      </li>
      <li>
        <NavLink to="/missions">
          Missions
        </NavLink>
      </li>
      <li>
        <NavLink to="/profile">
          My Profile
        </NavLink>
      </li>
    </ul>
  </nav>
);

export default Navbar;
