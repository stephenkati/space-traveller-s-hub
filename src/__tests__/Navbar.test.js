import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('tests for Navigation menu', () => {
  test('snapshot for Navbar component', () => {
    const component = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );

    expect(component).toMatchSnapshot();
  });
});
