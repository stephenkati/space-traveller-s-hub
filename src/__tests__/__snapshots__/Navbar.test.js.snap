// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Navigation menu snapshot for Navbar component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <div
          class="app-logo"
        >
          <img
            alt="logo"
            class="logo"
            src="planet.png"
          />
          <h1
            class="app-name"
          >
            Space Travelers’ Hub
          </h1>
        </div>
        <ul
          class="navLinks"
        >
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              class=""
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              class=""
              href="/profile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <div
        class="app-logo"
      >
        <img
          alt="logo"
          class="logo"
          src="planet.png"
        />
        <h1
          class="app-name"
        >
          Space Travelers’ Hub
        </h1>
      </div>
      <ul
        class="navLinks"
      >
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Rockets
          </a>
        </li>
        <li>
          <a
            class=""
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            class=""
            href="/profile"
          >
            My Profile
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
